packages/common/src/core/HubItemEntity.ts

```diff
@@ -56,8 +57,10 @@ export abstract class HubItemEntity<T extends IHubItemEntity>
     IWithPermissionBehavior,
     IWithDiscussionsBehavior,
     IWithFollowersBehavior,
-    IWithAssociationBehavior
+    IWithAssociationBehavior,
+    IWithEntitySettings
 {
+  entitySettings: Omit<IEntitySetting, "id"> & { id?: string };
   protected context: IArcGISContext;
```

Comment:
We need to make a couple updates to the `HubDiscussion` class methods to pass `context.hubRequestOptions` vs `context.requestOptions` or `context.userRequestOptions` so the calls through to the settings API don't error with an `Invalid token` response.

```jsx
// src/discussions/HubDiscussion.ts

  // ...

  static async fetch(
    identifier: string,
    context: IArcGISContext
  ): Promise<HubDiscussion> {
    try {
      const entity = await fetchDiscussion(identifier, context.hubRequestOptions);
      // create an instance of HubDiscussion from the entity
      return HubDiscussion.fromJson(entity, context);
    } catch (ex) {
      throw (ex as Error).message ===
        "CONT_0001: Item does not exist or is inaccessible."
        ? new Error("Discussion not found.")
        : ex;
    }
  }

  // ...

  async save(): Promise<void> {
    if (this.isDestroyed) {
      throw new Error("HubDiscussion is already destroyed.");
    }

    if (this.entity.id) {
      const { updateDiscussion } = await import("./edit");
      // update it
      this.entity = await updateDiscussion(
        this.entity,
        this.context.hubRequestOptions,
      );
    } else {
      const { createDiscussion } = await import("./edit");
      // create it
      this.entity = await createDiscussion(
        this.entity,
        this.context.hubRequestOptions,
      );
    }

    // call the after save hook on superclass
    await super.afterSave();

    return;
  }

  async delete(): Promise<void> {
    if (this.isDestroyed) {
      throw new Error("HubDiscussion is already destroyed.");
    }
    const { deleteDiscussion } = await import("./edit");
    this.isDestroyed = true;
    // Delegate to module fn
    await deleteDiscussion(this.entity.id, this.context.hubRequestOptions);
  }
```